core/block_svg.js

```diff
@@ -792,6 +792,28 @@ Blockly.BlockSvg.prototype.showContextMenu_ = function(e) {
     if (this.isEditable() && this.workspace.options.comments) {
       menuOptions.push(Blockly.ContextMenu.blockCommentOption(block));
     }
+
+    // pxt-blockly: only allow collapse on top-level event handler blocks
+    if (this.workspace.options.collapse
+        && block.previousConnection == null && block.outputConnection == null) {
```

Comment:
can previousConnection be undefined?

Reply:
undefined == null is true right?

Reply:
i find !block.previousConnection less confusing.